### PR TITLE
add support for desktop examples on MacOS (Apple Silicon M1=arm64/ x86_64/ i386)

### PR DESCRIPTION
**Pull request by walkoncross**
1. add config "build:darwin_arm64" in .bazelrc;
2. add "Build on MacOS" section in README.rd;
3. add ‘config_setting(name = "macos_arm64", …)’ in mediapipe/BUILD, and set default apple_platform_type
to ‘macos_x86_64’;
4. add build_macos_desktop_examples.sh

### CODE REVIEW

**Comment by walkoncross** [comment]
@googlebot I signed it!

File: README.md [hunk 1]
```diff
@@ -100,6 +100,71 @@ The source code is hosted in the
...
+```
+and run:
+```bash
+export GLOG_logtostderr=1
+bazel-bin/mediapipe/examples/desktop/hello_world/hello_world
+```
+
+2. Build all available examples under mediapipe/examples/desktop:
+```bash
+bash build_macos_desktop_examples.sh --cpu x86_64 -b
```
Comment (mgebhard) [inline README.md:157]: Super helpful I finally got helloworld to build on my m1 but any help with:

ERROR: Documents/mediapipe/mediapipe/framework/formats/BUILD:178:11: Compiling mediapipe/framework/formats/image_frame_opencv.cc failed: (Aborted): wrapped_clang failed: error executing command external/local_config_cc/wrapped_clang '-D_FORTIFY_SOURCE=1' -fstack-protector -fcolor-diagnostics -Wall -Wthread-safety -Wself-assign -fno-omit-frame-pointer -g0 -O2 -DNDEBUG '-DNS_BLOCK_ASSERTIONS=1' ... (remaining 64 argument(s) skipped)

Use --sandbox_debug to see verbose messages from the sandbox
In file included from mediapipe/framework/formats/image_frame_opencv.cc:15:
In file included from ./mediapipe/framework/formats/image_frame_opencv.h:20:
./mediapipe/framework/port/opencv_core_inc.h:18:10: fatal error: 'opencv2/core/version.hpp' file not found
#include <opencv2/core/version.hpp>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
Error in child process '/usr/bin/xcrun'. 1
Target //mediapipe/examples/desktop/face_detection:face_detection_cpu failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 1.063s, Critical Path: 0.92s
INFO: 46 processes: 36 internal, 10 darwin-sandbox.
FAILED: Build did NOT complete successfully

Comment (terasum) [reply README.md:157]: @mgebhard try modifiy the third_party/opencv_macos.BUILD :

```
--- a/third_party/opencv_macos.BUILD
+++ b/third_party/opencv_macos.BUILD
@@ -19,8 +19,8 @@ cc_library(
             "lib/libopencv_videoio.dylib",
         ],
     ),
-    hdrs = glob(["include/opencv2/**/*.h*"]),
-    includes = ["include/"],
+    hdrs = glob(["include/opencv4/opencv2/**/*.h*"]),
+    includes = ["include/opencv4/"],
     linkstatic = 1,
     visibility = ["//visibility:public"],
 )
```

**Comment by a-hurst** [comment]
@walkoncross Thanks a ton for this! Hopefully it gets merged soon.

I'm never really used Bazel before so no please ignore if I'm off-base, but based on the surrounding code couldn't you change the `WORKSPACE` `new_local_repository` lines to:
```
new_local_repository(
    name = "macos_opencv",
    build_file = "@//third_party:opencv_macos.BUILD",
    path = select({
        "//mediapipe:macos_arm64": "/opt/homebrew/opt/opencv@3",
        "//conditions:default": "/usr/local/opt/opencv@3",
    }),
)

new_local_repository(
    name = "macos_ffmpeg",
    build_file = "@//third_party:ffmpeg_macos.BUILD",
    path = select({
        "//mediapipe:macos_arm64": "/opt/homebrew/opt/ffmpeg",
        "//conditions:default": "/usr/local/opt/ffmpeg",
    }),
```
so you don't need to edit that file manually? You've already defined the `macos_arm64` platform type in `BUILD` so you should presumably be able to use it.

**Comment by cansik** [comment]
Would be great if mediapipe releases M1 packages officially. I managed to create a fully automated script to build version `0.8.9` with python wheels. I link opencv@3 and later delocate the wheel paths. The prebuilt wheels for version `3.8`, `3.9` and `3.10` are already uploaded and after some testing I will release them on PyPi as well. Looking forward to feedback and suggestions.

Here the repository with the build: https://github.com/cansik/mediapipe-silicon

**Comment by a-hurst** [comment]
> Would be great if mediapipe releases M1 packages officially. I managed to create a fully automated script to build version `0.8.9` with python wheels. I link opencv@3 and later delocate the wheel paths. The prebuilt wheels for version `3.8`, `3.9` and `3.10` are already uploaded and after some testing I will release them on PyPi as well. Looking forward to feedback and suggestions.
> 
> Here the repository with the build: https://github.com/cansik/mediapipe-silicon

Fantastic, thanks for putting this together! Will vastly simplify the setup and install process for anything that depends on mediapipe. For my use-case (extracting the x/y coordinates of face regions for cognitive face processing research), I can't realistically expect colleagues to set up and troubleshoot a compiler environment just to run a script, so having pre-built wheels is extremely useful.

**Comment by schmidt-sebastian** [comment]
Thank you for your contribution! We have integrated the required changes from your PR. We now support M1 for iOS and Android builds (Python support is available via Rosetta).

**Pull request closed without merging**